cartography/intel/github/repos.py

```diff
@@ -235,58 +244,123 @@ def _transform_collaborators(collaborators: Dict, repo_url: str, transformed_col
+    :param req_file_contents: Dict: The contents of the requirements.txt file.
     :param repo_url: str: The URL of the GitHub repo.
     :param out_requirements_files: Output array to append transformed results to.
     :return: Nothing.
     """
     if req_file_contents and req_file_contents.get('text'):
         text_contents = req_file_contents['text']
+        reqs_list = text_contents.split("\n")
+        _transform_python_requirements(reqs_list, repo_url, out_requirements_files)
 
-        parsed_list = []
-        for line in text_contents.split("\n"):
-            # Remove trailing comments and extra whitespace
-            stripped_line = line.partition('#')[0].strip()
-            if stripped_line == '':
-                continue
-            try:
-                req = Requirement(stripped_line)
-            except InvalidRequirement:
-                # INFO and not WARN/ERROR as we intentionally don't support all ways to specify Python requirements
-                logger.info(
-                    f"Failed to parse line \"{line}\" in repo {repo_url}'s requirements.txt; skipping line.",
-                    exc_info=True,
-                )
-                continue
-            parsed_list.append(req)
-
-        for req in parsed_list:
-            pinned_version = None
-            if len(req.specifier) == 1:
-                specifier = next(iter(req.specifier))
-                if specifier.operator == '==':
-                    pinned_version = specifier.version
-
-            # Set `spec` to a default value. Example values for str(req.specifier): "<4.0,>=3.0" or "==1.0.0".
-            spec: Optional[str] = str(req.specifier)
-            # Set spec to `None` instead of empty string so that the Neo4j driver will leave the library.specifier field
-            # undefined. As convention, we prefer undefined values over empty strings in the graph.
-            if spec == '':
-                spec = None
-
-            canon_name = canonicalize_name(req.name)
-            requirement_id = f"{canon_name}|{pinned_version}" if pinned_version else canon_name
-
-            out_requirements_files.append({
-                "id": requirement_id,
-                "name": canon_name,
-                "specifier": spec,
-                "version": pinned_version,
-                "repo_url": repo_url,
-            })
+
+def _transform_setup_cfg_requirements(
+    setup_cfg_contents: Dict,
+    repo_url: str,
+    out_requirements_files: List[Dict],
+) -> None:
+    """
+    Performs data transformations for the setup.cfg file in a GitHub repo, if available.
+    :param setup_cfg_contents: Dict: Contains contents of a repo's setup.cfg file.
+    :param repo_url: str: The URL of the GitHub repo.
+    :param out_requirements_files: Output array to append transformed results to.
+    :return: Nothing.
+    """
+    if setup_cfg_contents and setup_cfg_contents.get('text'):
```

Comment:
optional and changed the function argument, good catch thanks!

Reply:
if you want job security through obfuscation...

`if (text_contents := (setup_cfg_contents or {}).get('text')) is not None:`

jk that's not even valid since we're not guaranteed to be running under python3.8

Reply:
nit: prefer an early return so you can eliminate a level of indentation. i.e.
```
if not setup_cfg_contents or 'text' not in setup_cfg_contents:
    return
text_contents = setup_cfg_contents['text']
...
```